dataactbroker/permissions.py

```diff
@@ -31,21 +24,11 @@ def decorated_function(*args, **kwargs):
             try:
                 sess = GlobalDB.db().session
                 error_message = "Login Required"
-                if permission == "check_email_token":
-                    if LoginSession.isRegistering(session):
-                        return f(*args, **kwargs)
-                    else :
-                        error_message = "unauthorized"
-                elif permission == "check_password_token":
-                    if LoginSession.isResetingPassword(session):
-                        return f(*args, **kwargs)
-                    else :
-                        error_message  = "unauthorized"
```

Comment:
Not quite -- I just moved the functionality to the respective routes. We only used each bit of logic in one place.